TKGame/MainMenu.cs

```diff
@@ -0,0 +1,150 @@
+            {
+                LevelEditor.SaveStageDataToJSON(game.currentStage, "auto_saved_stage_data");
+                game.Exit();
+            };             
+        }
+
+        public static void LoadContent()
+        {
+            mainMenuGrid.ColumnsProportions.Add(new Proportion() { Type = ProportionType.Part });
+            
+            for (int i = 0; i < 2; i++)
+            {
+                mainMenuGrid.RowsProportions.Add(new Proportion() { Type = ProportionType.Part });
+            }
+
+            mainMenuGrid.Widgets.Add(playButton);
+            mainMenuGrid.Widgets.Add(exitButton);
+
+            desktop.Root = mainMenuGrid;
+        }
+
+        public static void Update()
+        {
+
+        }
+
+        public static void DisableMainMenu()
+        {
+            foreach (var widget in mainMenuGrid.Widgets)
+            {
+                widget.Enabled = false;
+                widget.Visible = false;
+            }
+        }
+
+        private static TextButton ConstructMenuButton(string text, string styleName, int gridRow, int gridCol, int width)
+        {
+            TextButton newButton = new TextButton();
+            newButton.Id = styleName;
+            newButton.Text = text;
+            newButton.GridColumn = gridCol;
+            newButton.GridRow = gridRow;
+            newButton.Width = width;
+            newButton.Height = 100;
+            newButton.HorizontalAlignment = HorizontalAlignment.Center;
+            newButton.VerticalAlignment = VerticalAlignment.Center;
+            newButton.Font = mainMenuFontSystem.GetFont(72);
+            newButton.Background = new SolidBrush(MAIN_MENU_BACKGROUND_COLOR);
+            newButton.TextColor = MAIN_MENU_TEXT_COLOR;
+            return newButton;
+        }
+
+        /// <summary>
+        /// Changes button text color to the original color when the mouse exits the button's area.
+        /// </summary>
+        /// <param name="o"></param>
+        /// <param name="e"></param>
+        private static void OnMouseLeaveButton(object obj, EventArgs e)
+        {
+            if (obj is not TextButton) throw new Exception("obj is not of type TextButton");
+
+            ((TextButton) obj).TextColor = MAIN_MENU_TEXT_COLOR;
+        }
+
+        /// <summary>
+        /// Changes button text color when mouse hovers over button. Color is determined from
+        /// textColorDict.
+        /// </summary>
+        /// <param name="obj"></param>
+        /// <param name="e"></param>
+        /// <exception cref="Exception"></exception>
+        private static void OnMouseEnterButton(object obj, EventArgs e) 
+        {
+            if (obj is not TextButton) throw new Exception("obj is not of type TextButton");
+            
+            TextButton b = (TextButton)obj;
+            Color c;
+
+            if(textColorDict.TryGetValue(b.Id, out c))
+                b.TextColor = c;
+        }
+    }
+}
```

Comment:
Could probably be refactored in the future but at least it's in its own class. Maybe create a main menu namespace for it as well for further separation.